panel_client/perfect_panel.py

```diff
@@ -0,0 +1,103 @@
+"""Demonstration/debugging panel. Uses the shell for I/O."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import Queue
+import threading
+from panel_client import PanelStateBase
+
+CONTROL_SCHEMES = [
+    [
+        {
+            'id': 'defenestrator',
+            'state': '0',
+            'actions': {
+                '0': '',
+                '1': 'Defenestrate the aristocracy!'
+            },
+            # HACK(jeff): `'type': 'button'` is only present to be able to play
+            # the game via the CLI, see where this is read at the bottom of
+            # this script.
+            'type': 'button',
+        },
+        {
+            'id': 'octo',
+            'state': 'nothing',
+            'actions': {
+                'nothing': '',
+                'nipple': 'Octo bite raven girl nipple!',
+                'mouth': 'Octo kiss raven girl mouth!'
+            },
+            'type': 'button'
+        },
+    ],
+    [{
+        'id': 'Froomulator',
+        'state': '0',
+
+        # This is a shorthand form of `states`, where the first value
+        # (the array) contains the values for `state`, and the second value
+        # (the string) is a "template action": the actual actions for each state
+        #  will be formed by replacing "%s" with the state.
+        'actions': [['0', '1', '2'], 'Set Froomulator to %s!']
+    }]
+]
+
+
+def poll_keyboard(stop_queue, input_queue, input_func):
+  print('polling started!')
+  while True:
+
+    try:
+      _ = stop_queue.get(block=False)
+      return
+    except Queue.Empty:
+      pass
+    try:
+      user_input = raw_input('Please specify (control_id state)')
```

Comment:
this will block until a line is present, and so won't look at `stop_queue` until that happens. if you want to improve over the original implementation, you can probably steal the implementation of `KeyboardReader` from https://github.com/wearhere/spacecontrol/pull/7/commits/acf10f46c22fcb754a0074569ef28056981c57b8 -- that implementation doesn't even try to read from `stdin` until a line is available there (`select` only returns `stdin` when there's an entire line available because `stdin` is line-buffered, a pretty unusual kind of file)

Reply:
although jeff and i agreed in the pr that poling the keyboard is just a placeholder

Reply:
Yeah I started with this because it's fast. Got partway through switching over to using select, but also wanted to sleep :)

Reply:
Now with 99% less blocking!